pdr_backend/analytics/predictoor_dashboard/dash_components/plots.py

```diff
@@ -0,0 +1,151 @@
+        name (str): Name of the trace.
+        x_data (list): List of x values.
+        y_data (list): List of y values.
+    Returns:
+        go.Bar: Bar plot trace.
+    """
+    return go.Bar(x=x_data, y=y_data, name=name, width=1)
+
+
+@enforce_types
+def process_payouts(payouts: List[dict], predictor: str, feed: str) -> tuple:
+    """
+    Process payouts data for a given predictor and feed.
+    Args:
+        payouts (list): List of payouts data.
+        predictor (str): Predictor address.
+        feed (str): Feed contract address.
+    Returns:
+        tuple: Tuple of slots, accuracies, profits, and stakes.
+    """
+    slots, accuracies, profits, stakes = [], [], [], []
+    profit = predictions = correct_predictions = 0
+
+    for p in payouts:
+        if predictor in p["ID"] and feed in p["ID"]:
+            predictions += 1
+            profit_change = p["payout"] - p["stake"] if p["payout"] > 0 else -p["stake"]
+            profit += profit_change
+            correct_predictions += p["payout"] > 0
+
+            slots.append(p["slot"])
+            accuracies.append((correct_predictions / predictions) * 100)
+            profits.append(profit)
+            stakes.append(p["stake"])
+    slot_in_date_format = [
+        UnixTimeS(ts).to_milliseconds().to_dt().strftime("%m-%d %H:%M") for ts in slots
+    ]
+    return slot_in_date_format, accuracies, profits, stakes
+
+
+@enforce_types
+def create_figure(
+    data_traces: List[Union[go.Scatter, go.Bar]],
+    title: str,
+    yaxis_title: str,
+    show_legend: bool = True,
+):
+    """
+    Create a figure with the given data traces.
+    Args:
+        data_traces (list): List of data traces.
+        title (str): Figure title.
+        yaxis_title (str): Y-axis title.
+        show_legend (bool): Show legend. Default is True.
+    Returns:
+        go.Figure: Plotly figure.
+    """
+    fig = go.Figure(data_traces)
+    fig.update_layout(
+        title=title,
+        yaxis_title=yaxis_title,
+        margin={"l": 20, "r": 0, "t": 50, "b": 0},
+        showlegend=show_legend,
+        xaxis_nticks=4,
+        bargap=0.1,
+        barmode="stack",
+        legend=(
+            {
+                "orientation": "h",
+                "yanchor": "bottom",
+                "y": 1.02,
+                "xanchor": "right",
+                "x": 1,
+            }
+            if show_legend
+            else {}
+        ),
+    )
+    return fig
+
+
+@enforce_types
+def get_figures(
+    payouts: Optional[List], feeds: List, predictoors: List[str]
+) -> Tuple[go.Figure, go.Figure, go.Figure]:
+    """
+    Get figures for accuracy, profit, and costs.
+    Args:
+        payouts (list): List of payouts data.
+        feeds (list): List of feeds data.
+        predictoors (list): List of predictoors data.
+    Returns:
+        tuple: Tuple of accuracy, profit, and costs figures.
+    """
+    accuracy_scatters, profit_scatters, stakes_scatters = [], [], []
+
+    if payouts:
```

Comment:
can we remove some imbrication here?
e.g. let's build the scatters and check if empty at the end